braindecode/datasets/base.py

```diff
@@ -194,7 +209,21 @@ def __getitem__(self, index):
         np.ndarray
             Crop indices.
         """
-        X = self.windows.get_data(item=index)[0].astype('float32')
+
+        # necessary to cast as list to get list of three tensors from batch,
+        # otherwise get single 2d-tensor...
+        crop_inds = self.crop_inds[index].tolist()
+
+        i_window_in_trial, i_start, i_stop = crop_inds
+        if hasattr(self.windows, 'metadata'):  # mne epochs case:
+            X = self.windows.get_data(item=index)[0].astype('float32')
+        else:
+            X = self.windows._getitem((slice(None), slice(i_start, i_stop)), return_times=False)
+        X = X.astype('float32')
```

Comment:
can this casting be done once for the whole raw file instead of casting at every `__getitem__`?

Reply:
Maybe we should be able to select precision? I imagine that someone may want to save memory and train model with fp16?

Reply:
and BTW, we are casting two times in the case of `hasattr(self.windows, 'metadata')==True`

Reply:
> can this casting be done once for the whole raw file instead of casting at every __getitem__?
I think not possible with mne unfortunately, at least that was my understanding. Otherwise of course would be better to be in float32 all the time 

> Maybe we should be able to select precision? I imagine that someone may want to save memory and train model with fp16?
Maybe in the future, although if purely for training speedup/GPU memory, could be done outside dataset I guess.

> and BTW, we are casting two times in the case of hasattr(self.windows, 'metadata')==True
Thanks, fixed